edalize/vcs.py

```diff
@@ -0,0 +1,86 @@
+import os
+import logging
+
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+class Vcs(Edatool):
+
+    tool_options = {
+        'lists' : {'vcs_options' : 'String'}
+    }
+
+    argtypes = ['plusarg', 'vlogdefine', 'vlogparam']
+
+    MAKEFILE_TEMPLATE = """
+all: $(TARGET)
+
+$(TARGET):
+	vcs -full64 -t $(TOPLEVEL) -f $(TARGET).scr -o $@ $(VCS_OPTIONS)
+
+run: $(TARGET)
+	$(TARGET) -l vcs.log $(EXTRA_OPTIONS)
+
+clean:
+	$(RM) $(TARGET)
+"""
+
+    def configure_main(self):
+        f = open(os.path.join(self.work_root, self.name+'.scr'),'w')
+
+        (src_files, incdirs) = self._get_fileset_files()
+        for key, value in self.vlogdefine.items():
+            f.write('+define+{}={}\n'.format(key, self._param_value_str(value, '')))
+
+        for key, value in self.vlogparam.items():
+            f.write('+parameter+{}.{}={}\n'.format(self.toplevel, key, self._param_value_str(value, '"')))
+        for id in incdirs:
+            f.write("+incdir+" + id+'\n')
+        timescale = self.tool_options.get('timescale')
+        if timescale:
+            with open(os.path.join(self.work_root, 'timescale.v'), 'w') as tsfile:
+                tsfile.write("`timescale {}\n".format(timescale))
+            f.write('timescale.v\n')
+        for src_file in src_files:
+            if src_file.file_type in ["verilogSource",
```

Comment:
I guess vcs support VHDL as well, but if you don't have any VHDL code to test with, this can be added later upon request